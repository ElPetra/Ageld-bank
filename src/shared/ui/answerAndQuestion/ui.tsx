import React, { useState } from 'react';

import { Icon } from '../icon';
import './style.scss';

interface Props {
    question: string;
    answer: string;
}

export const AnswerAndQuestion: React.FC<Props> = ({ question, answer }) => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleAnswer = () => {
        setIsOpen(prevState => !prevState);
    };

    return (
        <div className='block'>
            <div className='question'>
                <p className='question__text'>{question}</p>
                <button className='question__button' onClick={toggleAnswer}>
                    <Icon
                        width={24}
                        icon={isOpen ? 'arrow-up' : 'arrow-down'}
                    />
                </button>
            </div>
            {isOpen && (
                <div className='answer'>
                    <p className='answer__text'>{answer}</p>
                </div>
            )}
        </div>
    );
};
