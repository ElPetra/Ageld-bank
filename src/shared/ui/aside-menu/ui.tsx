import cn from 'classnames';

import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    visible: boolean;
    children: ReactNode;
}

export const AsideMenu = ({ visible, children }: Props) => {
    const asideClass = cn('aside-menu', {
        visible: visible
    });
    return (
        <>
            {visible && <div className='overlay' />}
            <aside className={asideClass}>
                <div className='aside-menu__content'>{children}</div>
            </aside>
        </>
    );
};
