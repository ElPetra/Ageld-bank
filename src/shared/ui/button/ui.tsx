import { memo } from 'react';

import cn from 'classnames';

import type { ButtonHTMLAttributes, ReactNode } from 'react';

import './styles.scss';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
    type?: 'submit' | 'reset' | 'button';
    disabled?: boolean;
    size?: 'medium' | 'large';
    variant?: 'primary' | 'secondary' | 'link';
    width?: 'auto' | 'max';
    status?: 'active' | undefined;
    children: ReactNode;
}

export const Button = memo(
    ({
        type,
        size = 'medium',
        disabled = false,
        variant = 'primary',
        width = 'auto',
        status,
        children,
        ...props
    }: Props) => {
        const buttonClass = cn(size, width, status, {
            button: variant !== 'link',
            [variant]: !disabled
        });
        return (
            <button
                type={type || 'button'}
                disabled={disabled}
                className={buttonClass}
                {...props}
            >
                {children}
            </button>
        );
    }
);
