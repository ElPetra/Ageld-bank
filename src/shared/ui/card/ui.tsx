import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    children: ReactNode;
    gap?: 'small' | 'medium' | 'large';
    direction?: 'column' | 'row';
    status?: 'active' | '';
}

export const Card = ({
    children,
    gap = 'medium',
    direction = 'column',
    status
}: Props) => (
    <div className={`card ${gap + '-gap'} ${direction} ${status}`}>
        {children}
    </div>
);
