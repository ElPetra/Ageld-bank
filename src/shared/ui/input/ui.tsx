import { InputHTMLAttributes, memo, ReactNode } from 'react';

import './styles.scss';

interface Props extends Omit<InputHTMLAttributes<HTMLInputElement>, 'size'> {
    type?: 'button' | 'checkbox' | 'color' | 'date' | 'password' | 'email' | 'file' | 'hidden' | 'image' | 'month' | 'number' | 'text';
    placeholder?: string;
    size?: 'medium' | 'large';
    width?: 'auto' | 'max';
    status?: 'active' | 'error' | undefined;
    children?: ReactNode;
}

export const Input = memo(({
                               type,
                               placeholder,
                               size = 'medium',
                               width = 'auto',
                               status,
                               children,
                               ...props
                           }: Props) =>
    <div className='field'>
        <span className='placeholder'>
                {placeholder}
            </span>
        <input type={type || 'text'} placeholder={placeholder || ''}
               className={`input ${status} ${size} ${width}`} {...props}>
            {children}
        </input>
    </div>
);

