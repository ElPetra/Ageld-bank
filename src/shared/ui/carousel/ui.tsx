import { useRef } from 'react';
import { useCarousel, useCarouselControls } from 'src/shared/lib/carousel';

import { CarouselInner } from './ui/inner';
import { CarouselIndicators } from './ui/indicators';
import { CarouselControls } from './ui/controls';

import type { ReactNode } from 'react';

import './styles.scss';

interface Props {
    cards: ReactNode[];
}

export const Carousel = ({ cards }: Props) => {
    const containerRef = useRef<HTMLDivElement>(null);
    const { angle, distance } = useCarousel(cards.length);
    const { changeRotateByIndicator, changeRotateByControl, active } =
        useCarouselControls(cards.length, containerRef, angle);
    return (
        <div className='carousel'>
            <div>
                <CarouselInner
                    distance={distance}
                    angle={angle}
                    active={active}
                    containerRef={containerRef}
                    cards={cards}
                />
                <CarouselIndicators
                    cards={cards}
                    changeRotateByIndicator={changeRotateByIndicator}
                    active={active}
                />
            </div>
            <CarouselControls changeRotateByControl={changeRotateByControl} />
        </div>
    );
};
