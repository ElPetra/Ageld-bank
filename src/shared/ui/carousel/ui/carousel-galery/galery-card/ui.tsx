import { Button, Card, Text } from 'src/shared/ui';
import './styles.scss';
import { type ReactNode } from 'react';

interface Props {
    title: string;
    content: string | ReactNode;
    button: string;
}

export const GalleryCard = ({ title = 'Новинка!', content, button }: Props) => {
    return (
        <Card
            className='gallery_card'
            justify='space-between'
            direction='column'
            gap='small'
            borderRadius='large'
            padding='medium'
        >
            <Text size='m' weight='bold' align='center'>
                {title}
            </Text>
            <Text size='m'>{content}</Text>
            <Button variant='secondary' size='medium' width='max'>
                {button}
            </Button>
        </Card>
    );
};
