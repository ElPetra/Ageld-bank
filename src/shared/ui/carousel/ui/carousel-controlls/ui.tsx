import cn from 'classnames';
import { type RefObject } from 'react';
import { useCarouselControlls } from 'src/shared/lib/carousel';
import './styles.scss';
import { useTranslation } from 'react-i18next';

interface Props {
    containerRef: RefObject<HTMLDivElement>;
    angle: number;
}

export const Controlls = ({ containerRef, angle }: Props) => {
    const { changeRotateByArrow, changeRotateByControl, active } =
        useCarouselControlls(containerRef, angle);
    const { t } = useTranslation();
    const controllClasses = (el: 'center' | 'prev' | 'next') =>
        cn({
            controll: true,
            active_controll: el === active
        });
    return (
        <div className='carousel_controlls_container'>
            <div className='controll_block_container'>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('prev')}
                >
                    <div
                        className={controllClasses('prev')}
                        aria-label={t('предыдущий слайд')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('center')}
                >
                    <div
                        className={controllClasses('center')}
                        aria-label={t('актуальный слайд')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByControl('next')}
                >
                    <div
                        className={controllClasses('next')}
                        aria-label={t('следующий слайд')}
                        role='button'
                    ></div>
                </div>
            </div>
            <div className='controll_block_container arrows'>
                <div
                    className='controll_block'
                    onClick={() => changeRotateByArrow('left')}
                >
                    <div
                        className='arrow controll'
                        aria-label={t('переключить назад')}
                        role='button'
                    ></div>
                </div>
                <div
                    className='controll_block right_arrow'
                    onClick={() => changeRotateByArrow('right')}
                >
                    <div
                        className='arrow controll'
                        aria-label={t('переключить вперед')}
                        role='button'
                    ></div>
                </div>
            </div>
        </div>
    );
};
