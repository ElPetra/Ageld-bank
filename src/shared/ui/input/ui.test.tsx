import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Input } from './ui';

describe('Input ui', () => {
    test('match snapshot', () => {
        render(<Input placeholder='Input' />);
        const input = screen.getByPlaceholderText('Input');
        expect(input).toMatchSnapshot();
    });

    test('label renders', () => {
        render(
            <Input
                placeholder='Input'
                size='large'
                value='value'
                onChange={() => {}}
            />
        );
        const input = screen.getByText('Input');
        expect(input).toBeInTheDocument();
    });

    test('label dont renders', () => {
        render(<Input placeholder='Input' />);
        const input = screen.queryByText('Input');
        expect(input).not.toBeInTheDocument();
    });

    test('error renders', () => {
        render(<Input error='Error' />);
        const input = screen.getByText('Error');
        expect(input).toBeInTheDocument();
    });

    test('error dont renders', () => {
        render(<Input error='Error' size='small' />);
        const input = screen.queryByText('Error');
        expect(input).not.toBeInTheDocument();
    });

    test('children renders', () => {
        render(<Input>Children</Input>);
        const input = screen.getByText('Children');
        expect(input).toBeInTheDocument();
    });

    test('onChange have been called', () => {
        const onChangeMock = jest.fn();
        render(<Input onChange={onChangeMock} placeholder='Input' />);
        const input = screen.getByPlaceholderText('Input');
        expect(input).toBeInTheDocument();
        fireEvent.change(input, { target: { value: 'Input1' } });
        expect(onChangeMock).toHaveBeenCalledTimes(1);
    });

    test('onChange have not been called', () => {
        const onChangeMock = jest.fn();
        render(
            <Input
                onChange={onChangeMock}
                placeholder='Input'
                disabled={true}
            />
        );
        const input = screen.getByPlaceholderText('Input');
        expect(input).toBeInTheDocument();
        fireEvent.change(input);
        expect(onChangeMock).toHaveBeenCalledTimes(0);
    });

    test('onBlur have been called', () => {
        const onBlurMock = jest.fn();
        render(<Input onBlur={onBlurMock} placeholder='Input' />);
        const input = screen.getByPlaceholderText('Input');
        expect(input).toBeInTheDocument();
        fireEvent.blur(input);
        expect(onBlurMock).toHaveBeenCalledTimes(1);
    });
});
