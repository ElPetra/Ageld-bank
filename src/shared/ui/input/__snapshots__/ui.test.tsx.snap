// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input ui match snapshot 1`] = `
<input
  class=""
  pattern=""
  placeholder="Input"
  type="text"
  value=""
/>
`;
